### DISCUSSION TITLE
Percentages

### DISCUSSION THREAD

**Discussion by DasLixou** [DioxusLabs/taffy#323] [Polls]
So i just got help from a discussion here and learned that any `::Percent` should be as values between `0.0` and `1.0`. (that explains why it is a f32 lol).

So I just want to "throw out" 2 ideas in the world:

1. Write a hint in any `::Percent` doc that it should be a value between `0.0` and `1.0`
2. Make `::Percent` contain `u8` between `0` and `100` instead of `f32`.

Both things would clear up confusion. Feedback welcome :)

**Comment by nicoburns** [comment]
Option 2 "Make ::Percent contain u8 between 0 and 100 instead of f32." won't work as:

- We need the ability to set decimal percentages (e.g. setting `width: 33.333333%` to take up 1/3rd of the space)
- Percentages above 100% are valid (e.g. `width: 200%`) although the use cases are fairly niche.

But we should definitely improve the documentation.

**Reply by DasLixou** [reply]
Ok that sounds like a good reason for Option 1
